app/components/work_packages/types/settings_component.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module WorkPackages
+  module Types
+    class SettingsComponent < ApplicationComponent
+      include OpPrimer::ComponentHelpers
+      include OpTurbo::Streamable
+
+      def initialize(model, **options)
+        @copy_workflow_from = options[:copy_workflow_from].to_i
```

Comment:
:green_circle: Maybe it's just me, but this `to_i` feels wrong to me. In the end it's just a matter of passing the value through from the outside, but maybe I'd prefer if the conversion took place outside of `SettingsComponent` (in the place that fills this from `params`, where we know that things are only _stringly_ typed.

Also as an aside: We _could_ make use of named arguments in this method. Something along the lines of:

```ruby
def initialize(model, copy_workflow_from: nil, **opts)
  # <--- use copy_workflow_from here
  super(model, **opts) # bonus: The superclass constructor will not receive the named argument that we consumed
end
```

Reply:
Maybe there is something wrong a bit deeper. When starting to develop this functionality I started using strings for the type id. from params we also get strings, other records use string as their id. but for some reason, the `type.id` is an integer.

so, I have two options
- sanitizing this when first reading from params
- have the comparison function work to safely compare strings with integer ids

WDYT?

Reply:
I'd prefer to sanitize when reading from `params`. The latter is what a Rails method would probably try to do (e.g. `Type.find(...)`. But if I can choose, I try to write code that uses the correct types early.